mindsdb/integrations/handlers/replicate_handler/replicate_handler.py

```diff
@@ -0,0 +1,143 @@
+import replicate
+import pandas as pd
+from mindsdb.integrations.libs.base import BaseMLEngine
+from typing import Dict, Optional
+import os
+import types
+from mindsdb.utilities.config import Config
+
+
+class ReplicateHandler(BaseMLEngine):
+    name = "replicate"
+
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+        if 'using' not in args:
+            raise Exception("Replicate engine requires a USING clause! Refer to its documentation for more details.")
+        else:
+            args = args['using']
+
+        if 'model_name' and 'version' not in args:
+            raise Exception('Add model_name and version ')
+
+            # Checking if passed model_name and version  are correct or not
+        try:
+            replicate.default_client.api_token = args['api_key']
+            replicate.models.get(args['model_name']).versions.get(args['version'])
+        except Exception as e:
+            raise Exception("Check your model_name and version carefully", e)
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """Saves model details in stirage to access it later
+        """
+        args = args['using']
+        args['target'] = target
+        self.model_storage.json_set('args', args)
+
+    def predict(self, df: pd.DataFrame, args: Optional[Dict] = None) -> pd.DataFrame:
+        """Using replicate makes the prediction according to your parameters
+        """
+        def filter_images(x):
+            output = replicate.run(
```

Comment:
(optional) I would suggest creating another helper function that calls `replicate.run`, as it doesn't align with the name `filter_images`. You could map both of them sequentially:

```
urls = map(get_urls, rows_arr)
filtered_urls = map(filter_images, urls)
```